src/ui/public/timepicker/kbn_global_timepicker.js

```diff
@@ -19,13 +20,33 @@ UiModules
   return {
     template: toggleHtml,
     replace: true,
-    link: ($scope, $el, attrs) => {
+    link: ($scope) => {
       listenForUpdates($rootScope);
 
       $rootScope.timefilter = timefilter;
       $rootScope.toggleRefresh = () => {
         timefilter.refreshInterval.pause = !timefilter.refreshInterval.pause;
       };
+
+      // travel forward in time based on the interval between from and to
+      $scope.forward = function () {
+        assign(timefilter.time, timeNavigation.stepForward(timefilter.getBounds()));
+      };
+
+      // travel backwards in time based on the interval between from and to
+      $scope.back = function () {
+        assign(timefilter.time, timeNavigation.stepBackward(timefilter.getBounds()));
+      };
+
+      // zoom out, doubling the difference between start and end, keeping the same time range center
+      $scope.zoomOut = function () {
+        assign(timefilter.time, timeNavigation.zoomOut(timefilter.getBounds()));
+      };
+
+      // zoom in, halving the difference between start and end, keeping the same time range center
+      $scope.zoomIn = function () {
+        assign(timefilter.time, timeNavigation.zoomIn(timefilter.getBounds()));
+      };
```

Comment:
But I think they might be better placed in `time_navigation.js`, because then they'll be near the logic, which is the part that really needs explaining.